From akpm@osdl.org Mon Dec  4 15:16:19 2006
Message-Id: <200612042315.kB4NFvEk008570@shell0.pdx.osdl.net>
From: Jesper Juhl <jesper.juhl@gmail.com>
Subject: PCI: Be a bit defensive in quirk_nvidia_ck804() so we don't risk dereferencing a NULL pdev.
To: greg@kroah.com
Cc: akpm@osdl.org, jesper.juhl@gmail.com, alan@redhat.com
Date: Mon, 04 Dec 2006 15:14:48 -0800

From: Jesper Juhl <jesper.juhl@gmail.com>

pci_get_slot() may return NULL if nothing was found.  quirk_nvidia_ck804()
does not check the value returned from pci_get_slot(), so it may end up
causing a NULL pointer deref.

Signed-off-by: Jesper Juhl <jesper.juhl@gmail.com>
Acked-by: Alan Cox <alan@redhat.com>
Signed-off-by: Andrew Morton <akpm@osdl.org>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>


---
 drivers/pci/quirks.c |    2 ++
 1 file changed, 2 insertions(+)

--- gregkh-2.6.orig/drivers/pci/quirks.c
+++ gregkh-2.6/drivers/pci/quirks.c
@@ -1743,6 +1743,8 @@ static void __devinit quirk_nvidia_ck804
 	 * a single one having MSI is enough to be sure that MSI are supported.
 	 */
 	pdev = pci_get_slot(dev->bus, 0);
+	if (!pdev)
+		return;
 	if (!msi_ht_cap_enabled(dev) && !msi_ht_cap_enabled(pdev)) {
 		printk(KERN_WARNING "PCI: MSI quirk detected. "
 		       "MSI disabled on chipset %s.\n",
